superflore/generators/bitbake/yocto_recipe.py

```diff
@@ -152,33 +321,171 @@ def get_recipe_text(self, distributor, license_text):
+                conf_file.write(
+                    '\n# See generated-recipes-<ROS_DISTRO>/packagegroups/')
+                conf_file.write('packagegroup-ros-world.bb ')
+                conf_file.write('for a list of the generated recipes.\n')
+                conf_file.write(
+                    yoctoRecipe.generate_multiline_variable(
+                        'ROS_SUPERFLORE_GENERATED_RECIPES_FOR_COMPONENTS',
+                        yoctoRecipe.generated_components))
+                conf_file.write('\n# Packages found in the <buildtool_depend>'
+                                + ' and <buildtool_export_depend> items, ie,'
+                                + ' ones for which a -native is built.\n')
+                conf_file.write(
+                    yoctoRecipe.generate_multiline_variable(
+                        'ROS_SUPERFLORE_GENERATED_BUILDTOOLS',
+                        yoctoRecipe.generated_native_recipes))
+                ok('Wrote {0}'.format(conf_file_path))
+        except Exception as e:
+            err("Failed to write conf {0} to disk!".format(conf_file_path))
+            raise e
+
+    @staticmethod
+    def generate_packagegroup_ros_world(basepath, distro, skip_keys=[]):
+        pkggrp_path = "{0}/generated-recipes-{1}/packagegroups/".format(
+            basepath, distro)
+        pkggrp_file_path = '{0}packagegroup-ros-world.bb'.format(pkggrp_path)
+        try:
+            make_dir(pkggrp_path)
+            with open(pkggrp_file_path, "w") as pkggrp_file:
+                pkggrp_file.write('# Generated by superflore -- DO NOT EDIT\n')
+                pkggrp_file.write('#\n')
+                pkggrp_file.write(
+                    '# Copyright 2019 Open Source Robotics Foundation\n')
+                pkggrp_file.write(
+                    '# Distributed under the terms of the BSD license\n')
+                pkggrp_file.write('\nDESCRIPTION = "All packages listed in ')
+                pkggrp_file.write('${ROS_DISTRO}-cache.yaml"\n')
+                pkggrp_file.write('LICENSE = "MIT"\n\n')
+                pkggrp_file.write('inherit packagegroup\n\n')
+                pkggrp_file.write('PACKAGES = "${PN}"\n\n')
+                pkggrp_file.write(yoctoRecipe.generate_multiline_variable(
+                    'RDEPENDS_${PN}', yoctoRecipe.generated_recipes
+                    - yoctoRecipe.generated_native_recipes))
+                pkggrp_file.write('\n# Allow the above settings to be'
+                                  + ' overridden.\n')
+                pkggrp_file.write('include ${ROS_LAYERDIR}/recipes-ros/'
+                                  + 'packagegroups/packagegroup-ros-world-'
+                                  + '${ROS_DISTRO}.inc\n\n')
+                pkggrp_file.write('inherit ros_superflore_generated\n')
+                pkggrp_file.write('inherit ros_${ROS_DISTRO}\n')
+
+                ok('Wrote {0}'.format(pkggrp_file_path))
+        except Exception as e:
```

Comment:
What kind of `Exception` is raised here if this fails to write to disk? I'd prefer to catch that one type instead of others.

Reply:
Done by catching more specific OSError